-- @author prabhd 
-- @created 2013-02-01 12:00:00 
-- @modified 2013-02-01 12:00:00 
-- @tags cte
-- @product_version gpdb: [4.3-],hawq: [1.1-]
-- @db_name world_db
-- @description test16a: CTE within a user-defined function
DROP TABLE IF EXISTS foo;
DROP TABLE
CREATE TABLE foo AS SELECT i as a, i+1 as b from generate_series(1,10)i;
psql:sql/cte_functest_28.sql:9: NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column(s) named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
SELECT 10
CREATE OR REPLACE FUNCTION cte_func3()
RETURNS SETOF int 
RETURNS NULL ON NULL INPUT
AS $$
    id = []
    rv = plpy.execute("SELECT * FROM (WITH v AS (SELECT a, b FROM foo WHERE b < 9)\
SELECT v1.a FROM v AS v1, v as v2 WHERE v1.a = v2.b)OUTERFOO ORDER BY 1", 5)
    for i in range(0,5):
        val = rv[i]["a"]
        id.append(val)
    return id
$$ LANGUAGE plpythonu READS SQL DATA;
CREATE FUNCTION
WITH v(a, b) AS (SELECT a,b FROM foo WHERE b < 5)
SELECT * from v where b in ( select * from cte_func3()) ORDER BY 1;
psql:sql/cte_functest_28.sql:25: ERROR:  plpy.SPIError: function cannot execute on segment because it accesses relation "public.foo" (plpython.c:4648)  (entry db rh55-qa03.sanmateo.greenplum.com:3511 pid=4217) (cdbdisp.c:1476)
DETAIL:  
	 Traceback (most recent call last):
  PL/Python function "cte_func3", line 4, in <module>
    SELECT v1.a FROM v AS v1, v as v2 WHERE v1.a = v2.b)OUTERFOO ORDER BY 1", 5)
PL/Python function "cte_func3"
